emcc.py

```diff
@@ -420,13 +420,25 @@ def standardize_setting_change(key, value):
 
     if value[0] == '@':
       if key not in DEFERRED_RESPONSE_FILES:
-        value = open(value[1:]).read()
+        filename = value[1:]
+        if not os.path.exists(filename):
+          exit_with_error('%s: file not found parsing argument: %s' % (filename, change))
+        value = open(filename).read()
     else:
       value = value.replace('\\', '\\\\')
     try:
       value = parse_value(value)
     except Exception as e:
       exit_with_error('a problem occured in evaluating the content after a "-s", specifically "%s": %s', change, str(e))
+
+    # Do some basic type checking by comparing to the existing settings.
+    # Sadly we can't do this generically in the SettingsManager since there are settings
+    # that so change types internally over time.
+    existing = getattr(shared.Settings, user_key, None)
+    if existing is not None:
+      # We only currently worry about lists vs non-lists.
+      if (type(existing) == list) != (type(value) == list):
+        exit_with_error('setting `%s` expects `%s` but got `%s`' % (user_key, type(existing), type(value)))
```

Comment:
is this always valid? e.g. for `PTHREAD_POOL_SIZE` we allow both an integer and a string.

Reply:
Yeah, this is why this only triggers when one or the other is a list.   There are no settings that can be a list and something else.

The main reason for this so that we get get crashes internally when we try to add `.append` or  `+=` on something we think is a list.

Reply:
I see, sgtm. I can't think of a case where we mix a list with another type.